### ISSUE TITLE
[REQUEST] More recipes!

### ISSUE THREAD

**Issue by Atracador** [NyaaCat/RPGitems-reloaded#108]
Hi there! I've always wanted to be able to craft item with different recipes, just like some default ones such as helmet, boots (you can do them from above or below) or mushroom stews. Would you be able to implement a way for making more than one recipe for the same item instead of remaking it with a different name? Thanks for reading! Have a nice day!! Awesome plugin btw :smile:

**Comment by Formula350** [comment]
A solution for the *"like some default ones such as helmet, boots (you can do them from above or below) or mushroom stews."* would simply be to add an option to make it `Shaped` or `Shapeless`. Perhaps even a clickable-item inside the `Recipe Menu`: Redstone Dust for `Shaped`, Glowstone Dust for `Shapeless`, which when clicked on it would then save *and* close the `Recipe Menu`.

Additionally, since we're on the topic of Recipes, if I could make my own suggestions...? We have used recipes in RPG for a long time, but very very long ago we discovered that RPGItems is not "specific" enough when it looks for items that are crafted. It seems to only look at the `Item_Name:DataValue` and `Item_Name:DamageValue`, ignoring `nbt.display.Name` and `nbt.display.Lore`, despite storing that all in the recipe:
```yaml
   - ==: org.bukkit.inventory.ItemStack
      type: GOLD_SWORD
      meta:
        ==: ItemMeta
        meta-type: UNSPECIFIC
        display-name: §0§0§0§0§0§0§0§6§9§lCelestial Bronze Sword
        lore:
        - §fOne handed               Sword§c§a§f§e§0§0§0§0
        - §f8-9 Damage
        - '§e§o"A sword made from Celestial '
        - '§e§oBronze capable of '
        - §e§odestroying any monsters."
        - Droppable
```
Unfortunately, in that recipe the sword can be ANY Gold Sword, as the "Celestial Bronze Sword" that was placed in the recipe (ironically, an RPGItem itself) is not looked for specifically.

This has caused problems in our efforts to make Upgrades where a specific "special" item is needed in *addition* to the previous RPGItem. For example long ago I discovered that InfernalMobs plugin would drop these Netherstars that surprisingly worked as a unique recipe ingredient. What it ended up being is `Nether_Star:10`, and that `DataValue` was what allowed it to be a "special" item for recipes. *(Unfortunately, Mojang decided to screw textures up in 1.8, so ever since then Nether_Star:10 has been an ugly pink-black block :sigh:)* If RPGItems recipe would look for the additional meta details when crafting, that would allow us to make a lot more recipes by having the same general recipe items be the same, but even just 1 that is the same but has a different `display-name`, it would craft something else. :)

**TL;DR** - Please make it so `Recipe`, at the very least, also looks at the `display-name:` meta when crafting items. (Either by default, or with a toggle option, maybe `/rpgitem <item name> toggleRecipeMeta`)

**Comment by Atracador** [comment]
@Formula350 I like your idea, it's not a feature i currently use or need but it's definitely something to be worked on to. I bet it would lag the crafting table a bit though, if not, it's a win win.

**Comment by Omicronrg9** [comment]
I´m with Formula350.I think RPG servers could use it for multiple pruposes like upgrades,leveling...something like that.Greetings,Omicronrg9.

**Comment by Librazy** [comment]
Custom recipe by plugin in 1.12/1.13+ is not recommend. Recipe support in RPGItems main plugin may be removed in future.
